src/OmniSharp/Api/OmnisharpController.cs

```diff
@@ -9,11 +12,23 @@ public partial class OmnisharpController
     {
         private readonly OmnisharpWorkspace _workspace;
         private readonly OmniSharpOptions _options;
+        private readonly IApplicationShutdown _applicationShutdown;
 
-        public OmnisharpController(OmnisharpWorkspace workspace, IOptions<OmniSharpOptions> optionsAccessor)
+        public OmnisharpController(OmnisharpWorkspace workspace, IOptions<OmniSharpOptions> optionsAccessor, IApplicationShutdown applicationShutdown)
```

Comment:
It would be awesome if you could move this into it's own controller/class (and own file) similar to [ProjectSystem](https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp/Api/v1/ProjectSystem/OmnisharpController.ProjectSystem.cs).

1) you then you wouldn't have to change all the tests by inserting the FakeApplicationShutdown).
2) It would then be _much_ easier to merge into the [Plugins branch](https://github.com/OmniSharp/omnisharp-roslyn/tree/plugins/mef) which is now one class per one endpoint.